docs/understanding-tracking-design/json-schema-type-casting-rules/index.md

```diff
@@ -0,0 +1,370 @@
+    &nbsp;&nbsp;"type": "integer",<br/>
+    &nbsp;&nbsp;"enum": [E1, E2, ...]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>BIGINT</code>
+</td>
+<td>
+maximum <code>E*</code> &gt; 2147483647
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "integer"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>BIGINT</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"multipleOf": B<br/>
+&#125;</pre>
+</td>
+<td>
+<code>INT</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "number",<br/>
+    &nbsp;&nbsp;"multipleOf": B<br/>
+&#125;</pre>
+</td>
+<td>
+<code>DECIMAL(36,2)</code>
+</td>
+<td>
+Only works for <code>B</code>=2
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": ["number", "integer"]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>DOUBLE</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "number"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>DOUBLE</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "boolean"<br/>
+&#125;</pre>
+</td>
+<td><code>BOOLEAN</code></td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"minimum": M,<br/>
+    &nbsp;&nbsp;"maximum": M<br/>
+&#125;</pre>
+</td>
+<td>
+<code>CHAR(M)</code>
+</td>
+<td>
+Where <code>M</code> is the same in maximum and maximum
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"format": "uuid"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>CHAR(36)</code>
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"format": "ipv6"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(39)</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"format": "ipv4"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(15)</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"format": "email"<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(255)</code>
+</td><td></td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"type": "string",<br/>
+    &nbsp;&nbsp;"maxLength": M<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(M)</code>
+</td>
+<td>
+<code>enum</code> is not defined
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"enum": ["E1"]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>CHAR(M))</code>
+</td>
+<td>
+Where <code>M</code> is the size of json.stringify("E1"). <br/>
+<code>E1</code> is only element
+</td>
+</tr>
+<tr>
+<td>
+<pre lang="json">&#123;<br/>
+    &nbsp;&nbsp;"enum": ["E1", "E2"]<br/>
+&#125;</pre>
+</td>
+<td>
+<code>VARCHAR(M)</code>
+</td>
+<td>Where <code>M</code> maximum size of <code>json.stringify("E*")</code></td>
```

Comment:
Similar to the comment above — I would move this note to the column where `M` is defined. If you follow this approach for all rows, we will be left with just a few things in the “notes” column. These will be mostly remarks on the overall logic (e.g. handling of NULLABLE) or limitations (e.g. `B` = 2), rather than definitions of “variables” in the other two columns.